task/generate-labels/0.1/generate-labels.yaml

```diff
@@ -0,0 +1,73 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: "konflux"
+  name: generate-labels
+spec:
+  description: |
+    Generate labels based on templates.
+
+    Usage may look like the following.
+
+    > - name: generate-labels
+    >   params:
+    >   - name: LABEL_TEMPLATES
+    >     value: '["release=$SOURCE_DATE_EPOCH", "build-date=$SOURCE_DATE"]'
+
+    The following environment variables are defined for use in LABEL_TEMPLATES
+
+    * ACTUAL_DATE - a date time string containing the time this task runs, formatted +'%Y-%m-%dT%H:%M:%S'
+    * ACTUAL_DATE_EPOCH - the timestamp at the time this task runs
+    * SOURCE_DATE - a date time string containing the provided source timestamp, formatted +'%Y-%m-%dT%H:%M:%S'
+    * SOURCE_DATE_EPOCH - the timestamp provided as a param meant to represent the timestamp at which the source was last modified
+
+  params:
+  - name: label-templates
+    description: A json array of templates that should be rendered and exposed as a list of labels
+    type: string
+  - name: source-date-epoch
+    description: A standardised environment variable for build tools to consume in order to produce reproducible output.
+    default: ""
+  results:
+    - name: labels
+      description: The rendered labels, rendered from the provided templates
```

Comment:
Can we make the `label-templates` param and `labels` result both arrays?

Reply:
JSON strings are not great for composability, and special values could mess up the JSON

Reply:
Yes - I didn't see how, but you found a way above. I'll do it.

Reply:
OK. Update in place now at 5da7b94f6ded2317d3fe1d9f755420900d067b80.